example-apps/chatbot-rag-app/README.md

```diff
@@ -15,216 +15,111 @@ curl https://codeload.github.com/elastic/elasticsearch-labs/tar.gz/main | \
+and retry.
 
-Make sure you have a `.env` file with all your variables, then run:
+### Run locally
 
-```sh
-docker run --rm --env-file .env chatbot-rag-app flask create-index
-```
+If you want to run this example with Python and Node.js, you need to do a few
+things listed in the [Dockerfile](Dockerfile).
 
-See "Ingest data" section under Running Locally for more details about the `flask create-index` command.
+**Make sure you have a `.env` file with all your variables**
 
-#### Run API and frontend
+#### Build the frontend
 
-You will need to set the appropriate environment variables in your `.env` file. See the `env.example` file for instructions.
+The web assets are in the [frontend](frontend) directory, and built with yarn.
 
-```sh
-docker run --rm -p 4000:4000 --env-file .env -d chatbot-rag-app
+```bash
+# Install and use a recent node, if you don't have one.
+nvm install --lts
+nvm use --lts
+# Build the frontend web assets
+(cd frontend; yarn install; REACT_APP_API_HOST=/api yarn build)
 ```
 
-Note that if you are using an LLM that requires an external credentials file (such as Vertex AI), you will need to make this file accessible to the container in the `run` command above. For this you can use a bind mount, or you can also edit the Dockerfile to copy the credentials file to the container image at build time.
-
-### Locally (for development)
-
-With the environment variables set, you can run the following commands to start the server and frontend.
-
-#### Pre-requisites
+#### Configure your python environment
 
-- Python 3.8+
-- Node 14+
+Before we can run the app, we need a working Python environment with the
+correct packages installed:
 
-#### Install the dependencies
-
-For Python we recommend using a virtual environment.
-
-_ℹ️ Here's a good [primer](https://realpython.com/python-virtual-environments-a-primer) on virtual environments from Real Python._
-
-```sh
-# Create a virtual environment
-python -m venv .venv
-
-# Activate the virtual environment
+```bash
+python3 -m venv .venv
 source .venv/bin/activate
-
-# Install Python dependencies
+# install dev requirements for pip-compile and dotenv
+pip install pip-tools "python-dotenv[cli]"
+pip-compile
 pip install -r requirements.txt
-
-# Install Node dependencies
-cd frontend && yarn && cd ..
 ```
 
-#### Ingest data
-
-You can index the sample data from the provided .json files in the `data` folder:
+#### Run the ingest command
 
-```sh
-flask create-index
+First, ingest the data into elasticsearch:
+```bash
+$ dotenv run -- flask create-index
+".elser_model_2" model not available, downloading it now
+Model downloaded, starting deployment
+Loading data from ./data/data.json
+Loaded 15 documents
+Split 15 documents into 26 chunks
+Creating Elasticsearch sparse vector store in http://localhost:9200
 ```
 
-By default, this will index the data into the `workplace-app-docs` index. You can change this by setting the `ES_INDEX` environment variable.
-
-##### Indexing your own data
```

Comment:
@joemcelroy so I think this since it applies regardless of how you run it.. Maybe should be in a different section, like `## Customizing the data` that is if we want to encourage them to do it ;)

Reply:
added back at the bottom, fixing dead links